src/plugins/discover/public/application/components/top_nav/open_options_popover.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { I18nStart } from 'kibana/public';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiSpacer, EuiLink, EuiText, EuiWrappingPopover } from '@elastic/eui';
+import { getServices } from '../../../kibana_services';
+import './open_options_popover.scss';
+
+let isOpen = false;
+
+interface OptionsPopoverProps {
+  onClose: () => void;
+  anchorElement: HTMLElement;
+}
+
+export function OptionsPopover(props: OptionsPopoverProps) {
+  const {
+    core: { uiSettings },
+    addBasePath,
+  } = getServices();
+  const isLegacy = uiSettings.get('doc_table:legacy');
+
+  const linkText = isLegacy
+    ? i18n.translate('discover.openOptionsPopover.switchToDataGridText', {
+        defaultMessage: 'Switch to data grid',
+      })
+    : i18n.translate('discover.openOptionsPopover.switchToLegacyText', {
+        defaultMessage: 'Switch to legacy table',
+      });
+
+  return (
+    <EuiWrappingPopover ownFocus button={props.anchorElement} closePopover={props.onClose} isOpen>
+      <div className="dscOptionsPopover">
+        <EuiLink href={addBasePath('/app/management/kibana/settings?query=Use legacy table')}>
+          {linkText}
+        </EuiLink>
+        <EuiSpacer size="s" />
+        <EuiText color="subdued" size="s">
+          <FormattedMessage
+            id="discover.topNav.openOptionsPopover.description"
+            defaultMessage="The new data grid layout includes better data sorting, drag-and-drop columns, and a full
+                screen view. Enable this option if you prefer to fall back to the legacy table."
+          />
```

Comment:
thx a lot for taking care of this 🙇 , so here's my first thought while playing with this: I think also the link description should be different depending on which state is selected, adding @gchaps for wording

Reply:
Perhaps: "The new data grid features faster loading, better comparison, a flyout window, enhanced options, and is going to replace the legacy table soon. Enable this option if you prefer to fall back to the legacy table."

@gchaps what say you?